docs/doc/14-sql-commands/20-query-syntax/02-query-with.md

```diff
@@ -1,8 +1,13 @@
 ---
 title: WITH
 ---
+import FunctionDescription from '@site/src/components/FunctionDescription';
 
-Databend supports common table expressions (CTEs) and allows you to use a WITH clause to define one or multiple named temporary result sets that are used by the query that follows. The "temporary" means that the result sets will be not permanently stored anywhere in the database schema. They act as temporary views that are only available to the query that follows.
+<FunctionDescription description="Introduced or updated: v1.2.38"/>
+
+Databend supports common table expressions (CTEs) with a WITH clause, allowing you to define one or multiple named temporary result sets used by the following query. The term "temporary" implies that the result sets are not permanently stored in the database schema. They act as temporary views only accessible to the following query.
+
+Databend introduces the "Materialized" parameter for CTEs, allowing you to decide whether a CTE should be materialized or remain temporary. It is recommended to use the "Materialized" parameter for CTEs involving complex computations or large datasets to improve query performance by storing results on a physical storage medium. However, for simpler CTEs with limited usages, using "Materialized" may not be necessary. The decision should consider the trade-offs between performance gains and storage utilization.
```

Comment:
databend decides whether or not to inline a CTE by using the `materialized` keyword, if `materialized`, the CTE will not be inlined, so it will be executed only once, and then the result will be stored in memory, reducing repetitive execution of the CTE. The trade-off is that if `materialized`, i.e. not inline CTE, it will reduce the chance of the whole query predicate being pushed down and join reordered, so the business needs to make a choice based on its own situation.